lib/src/create_report.dart

```diff
@@ -335,120 +302,142 @@ Future<ReportSection> _trustworthDependency(
-    issues.add(_Issue(
-        'Found no Flutter in your PATH. Could not determine the current Flutter version.'));
-    allowsCurrentFlutter = false;
-  } else {
-    final usesFlutter = pubspec.usesFlutter;
-    final flutterDartVersion = Version.parse(
-        (flutterVersions['dartSdkVersion'] as String).split(' ').first);
-    final allowsCurrentFlutterDart =
-        sdkConstraint?.allows(flutterDartVersion) ?? false;
-
-    if (!allowsCurrentFlutterDart) {
-      allowsCurrentFlutter = false;
-      issues.add(
-        _Issue(
-          'The current sdk constraint does not allow the dart version latest of Flutter ($flutterDartVersion)',
+  Future<_Subsection> sdkSupport() async {
+    final issues = <_Issue>[];
+    final currentSdkVersion = Version.parse(Platform.version.split(' ').first);
+    final sdkConstraint = pubspec.dartSdkConstraint;
+    final allowsCurrentSdk = sdkConstraint?.allows(currentSdkVersion) ?? false;
+    if (sdkConstraint == null) {
+      issues.add(_Issue('Pubspec.yaml does not have an sdk version constraint.',
+          suggestion: 'Try adding an sdk constraint to your pubspec.yaml'));
+    } else if (!allowsCurrentSdk) {
+      issues.add(_Issue(
+          'The current sdk constraint does not allow the latest Dart ($currentSdkVersion)',
           span: _tryGetSpanFromYamlMap(pubspec.environment, 'sdk'),
-        ),
-      );
+          suggestion: 'Try widening the upper boundary of the constraint.'));
+    }
+
+    final flutterVersions = toolEnvironment.runtimeInfo.flutterVersions;
+    if (isRunningEnd2EndTest || flutterVersions == null) {
+      issues.add(_Issue(
+          'Found no Flutter in your PATH. Could not determine the current Flutter version.'));
     } else {
-      if (usesFlutter) {
-        // TODO(sigurdm): this will not work well locally (installed version will
-        // not be latest). Perhaps we should query somewhere for the latest version.
-        final currentFlutterVersion =
-            Version.parse(flutterVersions['frameworkVersion'] as String);
-        final flutterConstraint = pubspec.flutterSdkConstraint;
-        if (!flutterConstraint.allows(currentFlutterVersion)) {
-          issues.add(
-            _Issue(
-              'The current flutter constraint does not allow the latest Flutter ($currentFlutterVersion)',
-              span: _tryGetSpanFromYamlMap(pubspec.environment, 'flutter'),
-            ),
-          );
-          allowsCurrentFlutter = false;
+      final usesFlutter = pubspec.usesFlutter;
+      final flutterDartVersion = Version.parse(
+          (flutterVersions['dartSdkVersion'] as String).split(' ').first);
+      final allowsCurrentFlutterDart =
+          sdkConstraint?.allows(flutterDartVersion) ?? false;
+
+      if (!allowsCurrentFlutterDart) {
+        issues.add(
+          _Issue(
+            'The current sdk constraint does not allow the dart version latest of Flutter ($flutterDartVersion)',
```

Comment:
Change to:

`The current SDK constraint does not allow the Dart version used by the latest stable Flutter ($flutterDartVersion)`